### DISCUSSION TITLE
OTEL Javaagent initially provided K8's container+ attributes, but after changing from an older version no longer get passed to the collector.

### DISCUSSION THREAD

**Discussion by rbcksantoro** [open-telemetry/opentelemetry-java-instrumentation#8531] [Q&A]
When i turn on the otel java agent debug i no longer see k8attributes past onto the collector from the agent itself. Is there a known issue here or a change in how i get these additional attributes sent on in order to properly perform logical structuring of my metric, trace, and streaming log data?

Attributes i no longer see:
k8s.pod.name
k8s.namespace.name
etc... 
i can't recall the precise attribute name for the container but this is missing as well .. 

It's my understanding from reading the documentation this information should be provided by default w/o additional configuration of the agent. Am i correct? Please let me know if further information is needed, and excuse if I've missed anything obvious as I'm a new user of OTEL and the java agent in general. Any assistance or recommendations would be greatly appreciated.

**Comment by laurit** [comment]
Do you know which agent version used to produce these?

**Reply by rbcksantoro** [reply]
I've gone through a bunch of versions recently but i believe it was either 1.16-1.19 range. I didn't notice until after upgrading to 1.2x+ as one of my co-workers pointed out we were no longer receiving these resource attributes ..

**Reply by laurit** [reply]
To be honest I don't think that agent has ever produced `k8s` attributes, are you sure you didn't get these from the collector? If you believe that it was the agent that added these attributes then please provide the exact version that produces them (please do verify that you actually get these attributes with that version).

**Reply by rbcksantoro** [reply]
I trust you would know better than I. 😄 I would ask if there is any extension that would provide this information? or do you have any suggestions how i can inject these metrics? (I already use the downward api to obtain these attributes in my container). Is there a way i could pass them onto the opentelemetry-javaagent as resource attributes at runtime?

**Reply by laurit** [reply]
Usually such attributes are provided by a `ResourceProvicer` you can write your own and package it as an agent extension, here is a sample https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/examples/extension/src/main/java/com/example/javaagent/DemoResourceProvider.java and here is the source code for some more resource providers https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation/resources/library

**Reply by rbcksantoro** [reply]
This is where I'm confused, the metrics mentioned in this link are the ones i was originally receiving. I never configured or setup anything specific when i was receiving them though, so I'm not sure what i need to do the get them back.

https://opentelemetry.io/docs/specs/otel/resource/semantic_conventions/k8s/

**Reply by laurit** [reply]
Not everything that is in spec is implemented for all languages. I mostly work on `opentelemetry-java-instrumentation` and I am reasonably sure that we have never had anything that would fill these. Not too long ago resource providers were moved from `opentelemetry-java` to `opentelemetry-java-instrumentation`, as far as I can tell there never was a kubernetes resource provider in `opentelemetry-java` either. If you are sure that the agent sent them at some point then please do track down the exact version that sends them.
As far as I understand collector can add these resource attributes, see https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/k8sattributesprocessor and https://opentelemetry.io/blog/2022/k8s-metadata/ Alternatively you could set these yourself using `OTEL_RESOURCE_ATTRIBUTES` environment variable https://opentelemetry.io/docs/specs/otel/resource/sdk/#specifying-resource-information-via-an-environment-variable
